trillian/integration/hammer.go

```diff
@@ -1087,12 +1091,28 @@ func HammerCTLog(cfg HammerConfig) error {
 		glog.Info(s.String())
 	})
 
-	for count := uint64(1); count < cfg.Operations; count++ {
-		if err := s.retryOneOp(ctx); err != nil {
-			return err
+	if cfg.TestDuration == 0 {
+		// Test termination is controlled by operation count.
+		for count := uint64(1); count < cfg.Operations; count++ {
+			if err := s.retryOneOp(ctx); err != nil {
+				return err
+			}
+		}
+		glog.Infof("%s: completed %d operations on log", cfg.LogCfg.Prefix, cfg.Operations)
+	} else {
+		// Test termination is controlled by a timeout. Use a separate context
+		// to avoid the last operation propagating an error when the context
+		// times out.
+		ctx2, cancel := context.WithTimeout(context.Background(), cfg.TestDuration)
+		defer cancel()
+
+		for ctx2.Err() == nil {
+			if err := s.retryOneOp(ctx); err != nil {
+				return err
+			}
 		}
+		glog.Infof("%s: completed test duration %v on log", cfg.LogCfg.Prefix, cfg.TestDuration)
```

Comment:
Could log the actual duration it ran for here if you make the `ctx` change above?